python/GafferArnoldTest/IECoreArnoldPreviewTest/RendererTest.py

```diff
@@ -412,33 +412,56 @@ def testOSLShaderComponentConnections( self ) :
-			self.assertEqual( arnold.AiNodeGetStr( swizzleRSource, "shadername" ), "Maths/MixColor" )
+			# OSL 1.10+ allows component level connections, in which case we will
+			# get a link to each channel directly.
+			component = arnold.c_int()
+			rSource = arnold.AiNodeGetLink( outputNode, "param_a.r", component )
+			if rSource :
+
+				self.assertEqual( arnold.AiNodeEntryGetName( arnold.AiNodeGetNodeEntry( rSource ) ), "osl" )
+				self.assertEqual( arnold.AiNodeGetStr( rSource, "shadername" ), "Maths/MixColor" )
+				self.assertEqual( component.value, 2 ) # r is driven by b
+
+				gSource = arnold.AiNodeGetLink( outputNode, "param_a.g", component )
+				self.assertReferSameNode( rSource, gSource )
+				self.assertEqual( component.value, 0 ) # g is driven by r
+
+				bSource = arnold.AiNodeGetLink( outputNode, "param_a.b", component )
+				self.assertReferSameNode( rSource, bSource )
+				self.assertEqual( component.value, 1 ) # b is driven by g
+
+			else :
+
+				# fallback to OSL 1.9 where MaterialX shaders will have been inserted
+				# to emulate component level connections.
+
+				pack = arnold.AiNodeGetLink( outputNode, "param_a" )
+				self.assertEqual( arnold.AiNodeEntryGetName( arnold.AiNodeGetNodeEntry( pack ) ), "osl" )
+				self.assertEqual( arnold.AiNodeGetStr( pack, "shadername" ), "MaterialX/mx_pack_color" )
+
+				swizzleR = arnold.AiNodeGetLink( pack, "param_in1" )
+				self.assertEqual( arnold.AiNodeEntryGetName( arnold.AiNodeGetNodeEntry( swizzleR ) ), "osl" )
+				self.assertEqual( arnold.AiNodeGetStr( swizzleR, "shadername" ), "MaterialX/mx_swizzle_color_float" )
+				self.assertEqual( arnold.AiNodeGetStr( swizzleR, "param_channels" ), "b" )
+				swizzleRSource = arnold.AiNodeGetLink( swizzleR, "param_in" )
+
+				swizzleG = arnold.AiNodeGetLink( pack, "param_in2" )
+				self.assertEqual( arnold.AiNodeEntryGetName( arnold.AiNodeGetNodeEntry( swizzleG ) ), "osl" )
+				self.assertEqual( arnold.AiNodeGetStr( swizzleG, "shadername" ), "MaterialX/mx_swizzle_color_float" )
+				self.assertEqual( arnold.AiNodeGetStr( swizzleG, "param_channels" ), "r" )
+				swizzleGSource = arnold.AiNodeGetLink( swizzleG, "param_in" )
+
+				swizzleB = arnold.AiNodeGetLink( pack, "param_in3" )
+				self.assertEqual( arnold.AiNodeEntryGetName( arnold.AiNodeGetNodeEntry( swizzleB ) ), "osl" )
+				self.assertEqual( arnold.AiNodeGetStr( swizzleB, "shadername" ), "MaterialX/mx_swizzle_color_float" )
+				self.assertEqual( arnold.AiNodeGetStr( swizzleB, "param_channels" ), "g" )
+				swizzleBSource = arnold.AiNodeGetLink( swizzleG, "param_in" )
+
+				self.assertReferSameNode( swizzleRSource, swizzleGSource )
+				self.assertReferSameNode( swizzleRSource, swizzleBSource )
+
+				self.assertEqual( arnold.AiNodeEntryGetName( arnold.AiNodeGetNodeEntry( swizzleRSource ) ), "osl" )
+				self.assertEqual( arnold.AiNodeGetStr( swizzleRSource, "shadername" ), "Maths/MixColor" )
```

Comment:
We don't need this branch do we? We removed the call to `convertOSLComponentConnections()` because Arnold has an up to date OSL already.

Reply:
Addressed in 538b860